cpp/include/cuspatial/experimental/detail/point_linestring_distance.cuh

```diff
@@ -83,34 +105,64 @@ void __global__ pairwise_point_linestring_distance(Cart2dItA points_first,
   for (auto idx = threadIdx.x + blockIdx.x * blockDim.x;
        idx < std::distance(linestring_points_first, thrust::prev(linestring_points_last));
        idx += gridDim.x * blockDim.x) {
-    auto offsets_iter =
-      thrust::upper_bound(thrust::seq, linestring_offsets_first, linestring_offsets_last, idx);
+    auto linestring_offsets_iter = thrust::upper_bound(
+      thrust::seq, linestring_part_offsets_first, linestring_part_offsets_last, idx);
+
     // Pointer to the last point in the linestring, skip iteration.
     // Note that the last point for the last linestring is guarded by the grid-stride loop.
-    if (offsets_iter != linestring_offsets_last and *offsets_iter - 1 == idx) { continue; }
-
-    auto point_idx = thrust::distance(linestring_offsets_first, thrust::prev(offsets_iter));
+    if (linestring_offsets_iter != linestring_part_offsets_last &&
+        *linestring_offsets_iter - 1 == idx) {
+      continue;
+    }
+
+    auto linestring_offsets_idx =
+      thrust::distance(linestring_part_offsets_first, thrust::prev(linestring_offsets_iter));
+
+    auto part_offset_iter = thrust::upper_bound(thrust::seq,
```

Comment:
Wrong name of iterator. Add some comments to explain. Explore factoring outa as utility functions for traversal.